kadi/commands/validate.py

```diff
@@ -133,30 +133,30 @@ def __init__(self, stop=None, days: float = 14, no_exclude: bool = False):
         self.start: CxoTime = self.stop - days * u.day
         self.no_exclude = no_exclude
 
+        # Get the exclude intervals from the google sheet along with any auto-generated
+        # ones. This creates self.exclude_intervals which is a Table. By virtue of `tlm`
+        # and `states` properties that get used, this also creates self.tlm and
+        # self.states.
+        self.add_exclude_intervals()
+
     def __init_subclass__(cls, **kwargs):
         super().__init_subclass__(**kwargs)
         if cls.state_name is not None:
             cls.subclasses.append(cls)
 
-    @property
+    @functools.cached_property
     def tlm(self):
-        if not hasattr(self, "_tlm"):
-            logger.info(
-                f"Fetching telemetry for {self.msids} between {self.start.date} and"
+        logger.info(
+            f"Fetching telemetry for {self.msids} between {self.start.date} and"
+            f" {self.stop.date}"
+        )
+        _tlm = get_telem_table(self.msids, self.start, self.stop)
```

Comment:
If we're moving away from using attributes directly to store these (via`CLS._attribute_names`) it feels a little odd to bother with the underscore names, as we tend not to use them generally.

Reply:
This is to not overload the method name itself. I.e. inside `def tlm(self)`, `tlm` is defined as the unbound method. A type checker or IDE will notice stuff like this.

Reply:
Right, I wasn't suggesting `tlm` as the replacement but fine to leave these with the current naming hinting at the property-ness of the internals.